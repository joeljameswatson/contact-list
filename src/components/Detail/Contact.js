import React from "react";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import { withStyles } from "@material-ui/core/styles";
import { withRouter, Link } from "react-router-dom";
import Padding from "components/shared/Padding";
import Spinner from "components/shared/Spinner";

const styles = {
  card: {
    maxWidth: 400,
    margin: "30px auto"
  },
  media: {
    height: 300,
    objectFit: "cover"
  }
};

const Contact = ({
  contact = {},
  classes,
  handleRequestDelete,
  loading,
  id
}) => {
  return (
    <Padding>
      <Card className={classes.card}>
        {contact.avatar_large && (
          <CardMedia
            className={classes.media}
            image={`${process.env.PUBLIC_URL}${contact.avatar_large}`}
          />
        )}
        <Spinner isVisible={loading} />
        {!loading && (
          <CardContent>
            <Typography gutterBottom variant="headline" component="h2">
              {contact.first_name} {contact.last_name}
            </Typography>
            <Typography component="p">Email: {contact.email}</Typography>
            <Typography component="p">Phone: {contact.phone}</Typography>
          </CardContent>
        )}

        <CardActions>
          <Link to={`/edit/${id}`}>
            <Button size="small" color="primary">
              Edit
            </Button>
          </Link>
          <Button size="small" color="primary" onClick={handleRequestDelete}>
            Delete
          </Button>
        </CardActions>
      </Card>
    </Padding>
  );
};

export default withRouter(withStyles(styles)(Contact));
