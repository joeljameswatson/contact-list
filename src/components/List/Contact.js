import { Link, withRouter } from "react-router-dom";

import Avatar from "@material-ui/core/Avatar";
import FaceIcon from "@material-ui/icons/Face";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import React from "react";
import styled from "styled-components";

const Container = styled.div`
  a {
    text-decoration: none;
  }
`;

const Contact = ({ contact, match }) => {
  const fullName = `${contact.first_name} ${contact.last_name}`;
  return (
    <Container>
      <Link to={`${match.url}/${contact.id}`}>
        <ListItem key={contact.id} dense button>
          {contact.avatar ? (
            <Avatar
              alt={fullName}
              src={`${process.env.PUBLIC_URL}${contact.avatar}`}
            />
          ) : (
            <Avatar>
              <FaceIcon />
            </Avatar>
          )}
          <ListItemText primary={fullName} />
        </ListItem>
      </Link>
    </Container>
  );
};

export default withRouter(Contact);
