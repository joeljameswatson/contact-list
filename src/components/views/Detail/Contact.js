import { Link, withRouter } from "react-router-dom";

import Button from "@material-ui/core/Button";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Padding from "components/shared/Padding";
import React from "react";
import Spinner from "components/shared/Spinner";
import Typography from "@material-ui/core/Typography";
import styled from "styled-components";

const Container = styled.section`
  .card {
    max-width: 400px;
    margin: 30px auto;
  }
  .media {
    height: 300px;
    object-fit: cover;
  }
`;

export const Contact = ({ contact = {}, handleRequestDelete, loading, id }) => {
  return (
    <Container>
      <Padding>
        <Card className="card">
          {contact.avatar_large && (
            <CardMedia
              className="media"
              image={`${process.env.PUBLIC_URL}${contact.avatar_large}`}
            />
          )}
          <Spinner isVisible={loading} />
          {!loading && (
            <CardContent>
              <Typography gutterBottom variant="headline" component="h2">
                {contact.first_name} {contact.last_name}
              </Typography>
              <Typography component="p">Email: {contact.email}</Typography>
              <Typography component="p">Phone: {contact.phone}</Typography>
            </CardContent>
          )}

          <CardActions>
            <Link to={`/edit/${id}`}>
              <Button size="small" color="primary" id="edit-contact">
                Edit
              </Button>
            </Link>
            <Button
              size="small"
              color="primary"
              onClick={handleRequestDelete}
              id="delete-contact"
            >
              Delete
            </Button>
          </CardActions>
        </Card>
      </Padding>
    </Container>
  );
};

export default withRouter(Contact);
