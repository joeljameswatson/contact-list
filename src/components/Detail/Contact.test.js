import { Contact } from "./Contact";
import { MemoryRouter } from "react-router-dom";
import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

test("renders a contact correctly", () => {
  const props = {
    id: 4,
    contact: {
      id: "4",
      first_name: "Marchall",
      last_name: "Abdee",
      email: "mabdee0@behance.net",
      phone: "2539464783",
      avatar: "/avatars/24.jpg",
      avatar_large: "/avatars/24.jpg"
    },
    loading: false,
    handleRequestDelete: () => null,
    classes: {}
  };

  const tree = renderer
    .create(
      <MemoryRouter>
        <Contact {...props} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders correctly when loading", () => {
  const props = {
    id: 4,
    contact: {
      id: "4",
      first_name: "Marchall",
      last_name: "Abdee",
      email: "mabdee0@behance.net",
      phone: "2539464783",
      avatar: "/avatars/24.jpg",
      avatar_large: "/avatars/24.jpg"
    },
    loading: true,
    handleRequestDelete: () => null,
    classes: {}
  };

  const tree = renderer
    .create(
      <MemoryRouter>
        <Contact {...props} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders correctly when no large avatar image", () => {
  const props = {
    id: 4,
    contact: {
      id: "4",
      first_name: "Marchall",
      last_name: "Abdee",
      email: "mabdee0@behance.net",
      phone: "2539464783",
      avatar: "/avatars/24.jpg",
      avatar_large: ""
    },
    loading: false,
    handleRequestDelete: () => null,
    classes: {}
  };

  const tree = renderer
    .create(
      <MemoryRouter>
        <Contact {...props} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Contact calls handleRequestDelete on clicking delete", () => {
  const props = {
    classes: {},
    handleRequestDelete: jest.fn()
  };
  const Container = shallow(<Contact {...props} />);
  Container.find("#delete-contact").simulate("click");
  expect(props.handleRequestDelete).toHaveBeenCalledTimes(1);
});
