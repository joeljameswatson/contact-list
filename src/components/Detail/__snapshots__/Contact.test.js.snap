// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a contact correctly 1`] = `
<div
  className="sc-bdVaJa dNIUDI"
>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
  >
    <div
      className="MuiCardMedia-root-29"
      style={
        Object {
          "backgroundImage": "url(\\"/avatars/24.jpg\\")",
        }
      }
    />
    <div
      className="MuiCardContent-root-32"
    >
      <h2
        className="MuiTypography-root-33 MuiTypography-headline-38 MuiTypography-gutterBottom-51"
      >
        Marchall
         
        Abdee
      </h2>
      <p
        className="MuiTypography-root-33 MuiTypography-body1-42"
      >
        Email: 
        mabdee0@behance.net
      </p>
      <p
        className="MuiTypography-root-33 MuiTypography-body1-42"
      >
        Phone: 
        2539464783
      </p>
    </div>
    <div
      className="MuiCardActions-root-59"
    >
      <a
        className="MuiCardActions-action-60"
        href="/edit/4"
        onClick={[Function]}
      >
        <button
          className="MuiButtonBase-root-87 MuiButton-root-61 MuiButton-text-63 MuiButton-textPrimary-64 MuiButton-flat-66 MuiButton-flatPrimary-67 MuiButton-sizeSmall-84"
          disabled={false}
          id="edit-contact"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-62"
          >
            Edit
          </span>
          <span
            className="MuiTouchRipple-root-90"
          />
        </button>
      </a>
      <button
        className="MuiButtonBase-root-87 MuiButton-root-61 MuiButton-text-63 MuiButton-textPrimary-64 MuiButton-flat-66 MuiButton-flatPrimary-67 MuiButton-sizeSmall-84 MuiCardActions-action-60"
        disabled={false}
        id="delete-contact"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-62"
        >
          Delete
        </span>
        <span
          className="MuiTouchRipple-root-90"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly when loading 1`] = `
<div
  className="sc-bdVaJa dNIUDI"
>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
  >
    <div
      className="MuiCardMedia-root-29"
      style={
        Object {
          "backgroundImage": "url(\\"/avatars/24.jpg\\")",
        }
      }
    />
    <div
      className="Spinner-root-31"
    >
      <div
        className="MuiCircularProgress-root-97 MuiCircularProgress-colorPrimary-100 MuiCircularProgress-indeterminate-99"
        role="progressbar"
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg-102"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle-103 MuiCircularProgress-circleIndeterminate-105"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
    <div
      className="MuiCardActions-root-59"
    >
      <a
        className="MuiCardActions-action-60"
        href="/edit/4"
        onClick={[Function]}
      >
        <button
          className="MuiButtonBase-root-87 MuiButton-root-61 MuiButton-text-63 MuiButton-textPrimary-64 MuiButton-flat-66 MuiButton-flatPrimary-67 MuiButton-sizeSmall-84"
          disabled={false}
          id="edit-contact"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-62"
          >
            Edit
          </span>
          <span
            className="MuiTouchRipple-root-90"
          />
        </button>
      </a>
      <button
        className="MuiButtonBase-root-87 MuiButton-root-61 MuiButton-text-63 MuiButton-textPrimary-64 MuiButton-flat-66 MuiButton-flatPrimary-67 MuiButton-sizeSmall-84 MuiCardActions-action-60"
        disabled={false}
        id="delete-contact"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-62"
        >
          Delete
        </span>
        <span
          className="MuiTouchRipple-root-90"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly when no large avatar image 1`] = `
<div
  className="sc-bdVaJa dNIUDI"
>
  <div
    className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
  >
    
    <div
      className="MuiCardContent-root-32"
    >
      <h2
        className="MuiTypography-root-33 MuiTypography-headline-38 MuiTypography-gutterBottom-51"
      >
        Marchall
         
        Abdee
      </h2>
      <p
        className="MuiTypography-root-33 MuiTypography-body1-42"
      >
        Email: 
        mabdee0@behance.net
      </p>
      <p
        className="MuiTypography-root-33 MuiTypography-body1-42"
      >
        Phone: 
        2539464783
      </p>
    </div>
    <div
      className="MuiCardActions-root-59"
    >
      <a
        className="MuiCardActions-action-60"
        href="/edit/4"
        onClick={[Function]}
      >
        <button
          className="MuiButtonBase-root-87 MuiButton-root-61 MuiButton-text-63 MuiButton-textPrimary-64 MuiButton-flat-66 MuiButton-flatPrimary-67 MuiButton-sizeSmall-84"
          disabled={false}
          id="edit-contact"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-62"
          >
            Edit
          </span>
          <span
            className="MuiTouchRipple-root-90"
          />
        </button>
      </a>
      <button
        className="MuiButtonBase-root-87 MuiButton-root-61 MuiButton-text-63 MuiButton-textPrimary-64 MuiButton-flat-66 MuiButton-flatPrimary-67 MuiButton-sizeSmall-84 MuiCardActions-action-60"
        disabled={false}
        id="delete-contact"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-62"
        >
          Delete
        </span>
        <span
          className="MuiTouchRipple-root-90"
        />
      </button>
    </div>
  </div>
</div>
`;
